src/plots/Mesh/MeshViewerEnginePluginInfo.C

```diff
@@ -205,3 +209,100 @@ MeshViewerEnginePluginInfo::GetMenuName() const
+//
+//  Arguments:
+//    atts      The attribute subject. 
+//
+//  Programmer: Mark C. Miller, Thu May 30 22:41:51 PDT 2019
+// ****************************************************************************
+#include <avtColorTables.h>
+
+void
+MeshViewerEnginePluginInfo::SetMeshColor(AttributeSubject *atts)
+{
+    MeshAttributes *meshAtts = (MeshAttributes *)atts;
+
+    if (meshAtts->GetMeshColorSource() == MeshAttributes::MeshRandom)
+    {
+
+        ColorAttribute c;
+        unsigned char rgb[3] = {0,0,0};
+        
+        // 
+        // Try and get the color for the colorIndex'th color in the distinct
+        // discrete color table or, failing that, default discrete color table.
+        //
+        avtColorTables *ct = avtColorTables::Instance();
+        if (ct->GetControlPointColor(ct->GetDefaultDiscreteColorTable(), colorIndex, rgb))
+        {   
+            c.SetRed(int(rgb[0]));
+            c.SetGreen(int(rgb[1]));
+            c.SetBlue(int(rgb[2]));
+        }
+        else if (ct->GetControlPointColor("distinct", colorIndex, rgb))
+        {
+            c.SetRed(int(rgb[0]));
+            c.SetGreen(int(rgb[1]));
+            c.SetBlue(int(rgb[2]));
+        }
+
+        meshAtts->SetMeshColor(c);
+
+        // Increment the color index.
+        colorIndex = (colorIndex + 1) % ct->GetNumColors();
+    }
+}
+
+// ****************************************************************************
+//  Method: MeshViewerEnginePluginInfo::SetOpaqueColor
+//
+//  Purpose:
+//    Sets the mesh color if the MeshRandom att is set to true. 
+//
+//  Arguments:
+//    atts      The attribute subject. 
+//
+//  Programmer: Mark C. Miller, Thu May 30 22:41:51 PDT 2019
+// ****************************************************************************
+#include <avtColorTables.h>
+
+void
+MeshViewerEnginePluginInfo::SetOpaqueColor(AttributeSubject *atts)
+{
+    MeshAttributes *meshAtts = (MeshAttributes *)atts;
+
+    if (meshAtts->GetOpaqueColorSource() == MeshAttributes::OpaqueRandom)
+    {
+        ColorAttribute c;
+        unsigned char rgb[3] = {0,0,0};
+        
+        // 
+        // Try and get the color for the colorIndex'th color in the distinct
+        // discrete color table or, failing that, default discrete color table.
+        //
+        avtColorTables *ct = avtColorTables::Instance();
+        if (ct->GetControlPointColor(ct->GetDefaultDiscreteColorTable(), colorIndex, rgb))
+        {   
+            c.SetRed(int(rgb[0]));
+            c.SetGreen(int(rgb[1]));
+            c.SetBlue(int(rgb[2]));
+        }
+        else if (ct->GetControlPointColor("distinct", colorIndex, rgb))
+        {   
+            c.SetRed(int(rgb[0]));
+            c.SetGreen(int(rgb[1]));
+            c.SetBlue(int(rgb[2]));
+        }
+
+        meshAtts->SetOpaqueColor(c);
+
+        // Increment the color index.
+        colorIndex = (colorIndex + 1) % ct->GetNumColors();
+    }
+}
+
```

Comment:
Is the code in the Plugin Info still needed? If so, should it use your new avtColorTable method?

Reply:
Looks like I failed to re-gen that. Sorry. Let me re-gen really quickly. No, it is not needed.

Reply:
Ok, I just pushed that fix. Sorry 'bout that.

Reply:
oops. no I didn't. I fixed the *same* problem left in Curve plot. Let me fix mesh plot too. Doh!

Reply:
Ok, pushed fixes for this too.